File: process_logo.py
from PIL import Image
import numpy as np

def process_logo():
    input_path = "C:/Users/AbinaM/Desktop/we-brand-cms/logo-original.png"
    output_path = "C:/Users/AbinaM/Desktop/we-brand-cms/logo-processed.png"
    
    # Open the image and convert to RGBA
    img = Image.open(input_path).convert("RGBA")
    data = np.array(img)
    
    # Calculate luminance or darkness
    # The logo is dark blue on a light background. 
    # Let's use the average RGB value. If it's dark, it belongs to the logo.
    r, g, b, a = data[:,:,0], data[:,:,1], data[:,:,2], data[:,:,3]
    
    # Calculate grayscale brightness (0 to 255)
    brightness = 0.2989 * r + 0.5870 * g + 0.1140 * b
    
    # The background is light (high brightness), the logo is dark (low brightness)
    # We want the alpha channel to be 255 where it's dark, and 0 where it's light.
    # Let's invert the brightness: 255 - brightness.
    # To make it crisp, we can apply a threshold or just use the inverted brightness as alpha.
    alpha = 255 - brightness
    
    # Enhance the contrast of the alpha mask
    alpha = np.clip((alpha - 50) * 2, 0, 255)
    
    # Set RGB to pure white (255, 255, 255)
    data[:,:,0] = 255
    data[:,:,1] = 255
    data[:,:,2] = 255
    
    # Assign the calculated alpha mask
    data[:,:,3] = alpha.astype(np.uint8)
    
    # Create the new image and save
    new_img = Image.fromarray(data)
    new_img.save(output_path, "PNG")
    print("Logo processed and saved to:", output_path)

if __name__ == "__main__":
    process_logo()

File: process_logo.js
import sharp from 'sharp';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

async function processLogo() {
  const inputPath = path.join(__dirname, 'logo-original.png');
  const outputPath = path.join(__dirname, 'logo-processed.png');

  try {
    // Read the image
    const { data, info } = await sharp(inputPath)
      .raw()
      .ensureAlpha()
      .toBuffer({ resolveWithObject: true });

    // The image has a light background and a dark blue logo.
    // We want to turn the dark parts into solid white, and the light parts into transparent.
    for (let i = 0; i < data.length; i += 4) {
      const r = data[i];
      const g = data[i + 1];
      const b = data[i + 2];
      
      // Calculate perceived brightness (luminance)
      const brightness = 0.299 * r + 0.587 * g + 0.114 * b;
      
      // If it's a light pixel (brightness > 200), make it transparent
      // If it's a dark pixel (brightness <= 200), make it white with full opacity
      // To add anti-aliasing, we can map brightness to alpha.
      // Brightness range: ~0 (dark blue) to ~255 (white/light gray)
      // We want alpha = 255 for brightness = 0, alpha = 0 for brightness = 255.
      let alpha = 255 - brightness;
      
      // Increase contrast of alpha
      alpha = Math.max(0, Math.min(255, (alpha - 50) * 2));
      
      // Set to solid white
      data[i] = 255;     // R
      data[i + 1] = 255; // G
      data[i + 2] = 255; // B
      data[i + 3] = alpha; // A
    }

    await sharp(data, {
      raw: {
        width: info.width,
        height: info.height,
        channels: 4
      }
    }).toFile(outputPath);
    
    console.log('✅ Logo processed successfully and saved as logo-processed.png');
  } catch (error) {
    console.error('Error processing logo:', error);
  }
}

processLogo();
